akka-slf4j/src/main/scala/akka/event/slf4j/Slf4jLogger.scala

```diff
@@ -63,7 +63,15 @@ class Slf4jLogger extends Actor with SLF4JLogging with RequiresMessageQueue[Logg
   val mdcAkkaUidAttributeName = "akkaUid"
 
   private def akkaAddress = context.system.asInstanceOf[ExtendedActorSystem].provider.addressString
-  private val akkaUid: String = context.system.asInstanceOf[ExtendedActorSystem].uid.toString
+  private def akkaUid: String = {
+    try {
+      context.system.asInstanceOf[ExtendedActorSystem].uid.toString
+    } catch {
+      case _: IllegalStateException =>
+        // when logging before generating the uid (transport binding when remoting is enabled)
+        ""
+    }
+  }
```

Comment:
I considered deferring startDefaultLoggers until after transport init, but then we would loose some valuable logging that may happen at startup of transport. This will skip uid in mdc before it is defined.

Reply:
Seems reasonable